tests/blockchain/eth/contracts/integration/test_integration.py

```diff
@@ -492,6 +495,31 @@ def test_all(testerchain, token, escrow, policy_manager, user_escrow_proxy):
     testerchain.wait_for_receipt(tx)
     assert policy_manager_v1 == policy_manager.functions.target().call()
 
+    # Upgrade the user escrow library
+    # Deploy the same contract as the second version
+    user_escrow_proxy_v2, _ = testerchain.interface.deploy_contract(
+        'UserEscrowProxy', token.address, escrow.address, policy_manager.address)
+    user_escrow_secret2 = os.urandom(SECRET_LENGTH)
+    user_escrow_secret2_hash = testerchain.interface.w3.sha3(user_escrow_secret2)
+    # Ursula and Alice can't upgrade library, only owner can
+    with pytest.raises((TransactionFailed, ValueError)):
+        tx = user_escrow_linker.functions\
+            .upgrade(user_escrow_proxy_v2.address, user_escrow_secret, user_escrow_secret2_hash) \
+            .transact({'from': alice1})
+        testerchain.wait_for_receipt(tx)
+    with pytest.raises((TransactionFailed, ValueError)):
+        tx = user_escrow_linker.functions\
+            .upgrade(user_escrow_proxy_v2.address, user_escrow_secret, user_escrow_secret2_hash) \
+            .transact({'from': ursula1})
+        testerchain.wait_for_receipt(tx)
+
+    # Upgrade library
+    tx = user_escrow_linker.functions\
+        .upgrade(user_escrow_proxy_v2.address, user_escrow_secret, user_escrow_secret2_hash) \
+        .transact({'from': creator})
+    testerchain.wait_for_receipt(tx)
+    assert user_escrow_proxy_v2.address == user_escrow_linker.functions.target().call()
+
     # Unlock and withdraw all tokens in MinersEscrow
     for index in range(11):
```

Comment:
Does the entire ethereum contract test suite assume 10 test accounts?  I'd like to make an effort to add more convention as to which accounts are used for what operations in tests.

Reply:
Only 10 by default.
What do you mean? Do you want to see comment for characters and accounts?